src/test/java/com/google/cloud/spanner/hibernate/GeneratedSelectStatementsTests.java

```diff
@@ -0,0 +1,115 @@
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+package com.google.cloud.spanner.hibernate;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.cloud.spanner.hibernate.util.SubTestEntity;
+import com.google.cloud.spanner.hibernate.util.TestEntity;
+import com.mockrunner.mock.jdbc.JDBCMockObjectFactory;
+import com.mockrunner.mock.jdbc.MockPreparedStatement;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import org.hibernate.Session;
+import org.hibernate.boot.Metadata;
+import org.hibernate.boot.MetadataSources;
+import org.hibernate.boot.registry.StandardServiceRegistry;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/**
+ * These tests use SpannerDialect and Hibernate-core to generate the final SELECT statements that
+ * are sent to the Spanner driver.
+ *
+ * @author Chengyuan
+ */
+public class GeneratedSelectStatementsTests {
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  private Metadata metadata;
+
+  private StandardServiceRegistry registry;
+
+  private JDBCMockObjectFactory jdbcMockObjectFactory;
+
+  /**
+   * Set up the metadata for Hibernate to generate schema statements.
+   */
+  @Before
+  public void setup() {
+    this.jdbcMockObjectFactory = new JDBCMockObjectFactory();
+    this.jdbcMockObjectFactory.registerMockDriver();
+    this.jdbcMockObjectFactory.getMockDriver()
+        .setupConnection(this.jdbcMockObjectFactory.getMockConnection());
+
+    this.registry = new StandardServiceRegistryBuilder()
+        .applySetting("hibernate.dialect", SpannerDialect.class.getName())
+        // must NOT set a driver class name so that Hibernate will use java.sql.DriverManager
+        // and discover the only mock driver we have set up.
+        .applySetting("hibernate.connection.url", "unused")
+        .applySetting("hibernate.connection.username", "unused")
+        .applySetting("hibernate.connection.password", "unused")
+        .build();
+    this.metadata =
+        new MetadataSources(this.registry).addAnnotatedClass(TestEntity.class)
+            .addAnnotatedClass(SubTestEntity.class).buildMetadata();
+  }
+
+  @Test
+  public void deleteDmlTest() {
+    testStatementTranslation(
+        x -> x.createQuery("delete TestEntity where boolVal = true").executeUpdate(),
```

Comment:
Assuming that we'll have more than one update test, you can make it even cleaner by writing a utility method
`testUpdateStatementTranslation("delete TestEntity where boolVal = true", "delete from test_table where boolVal=TRUE")`

Reply:
Yes , good point. will do.